source/common/phase_impl.h

```diff
@@ -0,0 +1,42 @@
+
+#pragma once
+
+#include "envoy/common/time.h"
+
+#include "nighthawk/common/phase.h"
+
+#include "external/envoy/source/common/common/logger.h"
+
+#include "absl/types/optional.h"
+
+namespace Nighthawk {
+
+class PhaseImpl : public Phase, public Envoy::Logger::Loggable<Envoy::Logger::Id::main> {
+public:
+  /**
+   * @param id Unique identifier of the pase (uniqueness not enforced).
+   * @param sequencer Sequencer that will be used to execute this phase.
+   * @param should_measure_latencies Indicates if latencies should be tracked for requests issued
+   * during execution of this phase.
+   * @param time_source Time source that will be used to query the clock.
+   * @param start_time Optional starting time of the phase. Can be used to schedule phases ahead.
```

Comment:
This is a re-implementation of something used today as a means to offset load test starts, to ensure request timings are evenly spaced from a global perspective. See https://github.com/envoyproxy/nighthawk/pull/219/files/c927a6ed9c006a0778813cf495c920408d1785bd#discussion_r363545616 for more context

Reply:
I'll leave this up to you and @mum4k for if that makes sense architecturally.  My concern is that we're defining an ordering mechanism directly into phases that doesn't seem to fit for phases, and we're repurposing it in ClientWorkerImpl for replacing a currently existing functionality that does not seem to have much to do with phases.

For whatever it's worth, that discussion link doesn't work, so I don't know what context you're referring to.

Reply:
The right link to the earlier discussion ought to have been https://github.com/envoyproxy/nighthawk/pull/219#discussion_r363545616. Somehow the previous one went bad, sorry. 
But point taken, somehow this doesn't fit well with me either, but so far I wasn't able to come up with something better. But reading back the linked discussion:

Maybe this should move into a different concept. I also had a concern with that this concept won't hold well with the new rate limiters we recently landed, and with non-probabilistic timing it doesn't make sense either. So maybe we should contain this functionality in a `DelayStartingRateLimiter` instead of here in `Phase`. How would you feel about that?

Reply:
I support your last suggestion @oschaaf, it makes the concept of phase more focused.

Reply:
Great; I added #281 as a prerequisite to this PR